[yong-zhuo] iP

yong-zhuo opened the pull request:
# Matcha 🍵 
> "Productivity is being able to do things that you were never able to do before." - Franz Kafka ([source](https://dansilvestre.com/productivity-quotes/))

**Matcha** helps you keep track of all your important tasks. It's,
- Text-based 📃
- ~Hard~ Easy to learn 🔰
- **SUPER FAST** to use ⚡

## Get Started:
1.  Download the JAR file [here](https://github.com/yong-zhuo/ip/releases/tag/A-Jar) 
2.  Run ```java -jar "{filename}.jar"``` in your terminal
3.  Start adding your tasks
4.  Let **Matcha** manage it for you! 

## Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

## Sample Code:
If you are a **Java** programmer, you can use it to practice **Java** too. Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yong-zhuo committed:
Rename Chatbot

yong-zhuo committed:
Add chatbot echo

yong-zhuo committed:
Add listing and adding tasks

yong-zhuo committed:
Add marking/unmarking tasks

yong-zhuo committed:
Add todo, event and deadline categories for tasks

yong-zhuo committed:
Add automated text UI testing

yong-zhuo committed:
Add error handling

yong-zhuo committed:
Add deleting tasks

yong-zhuo committed:
Rename and remove some functions

yong-zhuo committed:
Fix switch formatting

yong-zhuo committed:
Implement saving and loading tasks

yong-zhuo committed:
Merge branch 'branch-Level-7'

yong-zhuo committed:
Update chatbot to intepret date and time

yong-zhuo committed:
Merge branch 'branch-Level-8'

yong-zhuo committed:
Refactor code to include more OOP classes

yong-zhuo committed:
Merge branch 'branch-A-MoreOOP'

yong-zhuo committed:
Reorganise files into packages

yong-zhuo committed:
Reorganise more files into packages

yong-zhuo committed:
Merge branch 'branch-A-Packages'

yong-zhuo committed:
Merge branch 'add-gradle-support'

yong-zhuo committed:
Add JUnit test cases for Parser and Deadline classes

yong-zhuo committed:
Merge branch 'branch-A-JUnit'

yong-zhuo committed:
Edit Gradle file

yong-zhuo committed:
Merge branch 'branch-A-Jar'

yong-zhuo committed:
Add Javadoc comments to methods and classes

yong-zhuo committed:
Add finding task when given a keyword

yong-zhuo committed:
Update code to comply with coding standard

yong-zhuo committed:
Merge branch 'branch-A-JavaDoc'

yong-zhuo committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/matcha/task/Task.java
#	src/main/java/matcha/tasklist/TaskList.java

yong-zhuo committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/matcha/command/AddTaskCommand.java
#	src/main/java/matcha/task/Task.java
#	src/main/java/matcha/tasklist/TaskList.java
#	src/main/java/matcha/ui/Ui.java

yong-zhuo committed:
Fix Parser switch case formatting

yong-zhuo committed:
Add Checkstyle config files and fix coding standard violations

yong-zhuo committed:
Merge branch 'branch-A-CheckStyle'

yong-zhuo committed:
Add Main and Launcher classes

yong-zhuo committed:
Add images to resource folder

yong-zhuo committed:
Add GUI and modify Command classes

yong-zhuo committed:
Add CSS styling to application

yong-zhuo committed:
Add background image for app

yong-zhuo committed:
Merge branch 'branch-Level-10'

amoschee commented on src/main/java/matcha/command/Command.java:
```diff
@@ -0,0 +1,44 @@
+    private boolean isExit = false;
+
+    /**
+     * Sets the isExit boolean value to true.
```
Minor issue but you are missing the param description here

amoschee commented on src/main/java/matcha/storage/Storage.java:
```diff
@@ -0,0 +1,105 @@
+ * Represents a Storage object that saves and loads tasks from a file.
+ */
+public class Storage {
+    private static String FILE_PATH;
```
Since the variable is not final, perhaps it would be better to name it in camelCase instead

amoschee commented on src/test/java/matcha/parser/ParserTest.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    @Test
+    public void parseCommand_givenInput_returnsCorrectCommand() throws MatchaException {
```
Good job on testing extensively! 👍

amoschee reviewed (commented):
LGTM, just some minor issues regarding coding standard. Good job!

yong-zhuo committed:
Add application icon and fix coding standard issues

yong-zhuo committed:
Remove UI class

yong-zhuo committed:
Add assert statements to enhance code clarity for some classes

Currently, the code lacks explicit documentation for certain
important assumptions that should be made in the various classes.

This change was made to ensure that potential errors are caught
early and makes the code more robust.

Assert statements were integrated to document necessary assumptions
and improve overall readability.

yong-zhuo committed:
Refactor code for Parser and UpdateTaskCommand classes

There were some areas of code that could be simplified to enhance
readability.

Changes were made to the Parser and UpdateTaskCommand classes to
improve code comprehension for future debugging

The 'createParsedTask' method was extracted from the 'parseFileData'
method to enhance the readability of 'parseFileData'. The code in
the 'execute' method of UpdateTaskCommand was reorganised.

yong-zhuo committed:
Merge pull request #1 from yong-zhuo/branch-A-Assertions

Add assert statements to enhance code clarity for some classes

yong-zhuo committed:
Merge pull request #2 from yong-zhuo/branch-A-CodeQuality

Refactor code for Parser and UpdateTaskCommand classes

yong-zhuo committed:
Fix assert statements

Assert statements that were added to the various Command classes
was using the wrong condition for assertions.

Updated the conditions to correctly reflect the assumption made

yong-zhuo committed:
Merge pull request #3 from yong-zhuo/branch-A-Assertions

Fix assert statements

yu-sutong commented on src/main/java/matcha/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,136 @@
+/**
+ * Represents a command to add a task to the task list.
+ */
+public class AddTaskCommand extends Command {
```
Perhaps the name of a class should be a noun. (E.g. DeadlineCommandAdder)

yu-sutong commented on src/main/java/matcha/exception/MatchaException.java:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Represents a custom exception for all errors related to the Matcha program
+ */
+public class MatchaException extends Exception {
```
Should there be a more specific Exception to catch to avoid Pokemon Exception Handling?

yu-sutong commented on src/main/java/matcha/exception/MatchaException.java:
```diff
@@ -0,0 +1,24 @@
+     * Constructs a new MatchaException with the error message
+     * @param msg the error message
+     */
+    public MatchaException(String msg) {
```
Good attempt in incorporating the name of the chatbot to classes' names. However, it may be confusing to some who may not be sure that Meowception is an Exception.

yu-sutong reviewed (commented):
LGTM.

yong-zhuo committed:
Add feature to detect duplicate tasks when adding tasks

Currently, the bot is unable to check whether the user has added
a duplicate task, which could cause a hassle for the user to
manage their tasks.

It is important that we detect and prevent duplicates from existing
in the task list to prevent unintended actions.

To solve this, Task and its subclasses will now have a
'checkDuplicate' function to check if the tasks are similar.
When the user enters a duplicate task, the command will initiate
a check against all tasks in the task list to check if there are
any duplicates before allowing the user to add the task.
If duplicates are detected, the bot will inform the user.

This solution ensures that only unique tasks will be stored in the
task list for users.

yong-zhuo committed:
Merge branch 'branch-C-DetectDuplicates'

yong-zhuo committed:
Add Ui picture and edit README

Currently, there needs to be a user guide for the user to
understand how to use the application.

yong-zhuo committed:
Fix bug for detecting duplicate task

Previously, the user was unable to add a new task after a task is
added as the `checkDuplicate` function had a bug.

yong-zhuo committed:
Merge branch 'branch-C-DetectDuplicates'

yong-zhuo committed:
Merge branch 'master' into branch-A-UserGuide

yong-zhuo committed:
Finish editing README

User Guide is now updated to include more possible commands the
user might use

yong-zhuo committed:
Merge branch 'branch-A-UserGuide'

yong-zhuo committed:
Edit JavaDocs for Tasklist

Update header comment for clarity regarding the 'checkDuplicates'
function in TaskList

yong-zhuo committed:
Add command summary for user guide

A command summary will help users to quickly understand the
structure of the command while they are using the app

yong-zhuo committed:
Fix bugs when running JAR file

When running JAR file, an exception is thrown when trying to access
the app's icon.

There was a mistake made regarding the file path, changing it
fixed the app and the JAR file can be run successfully
